### PR TITLE
Update Yaml writer

### PR DESCRIPTION
**Pull request by kevmoo**
Expand the number of strings we can encode without quotes
if the value of a key/value pair is null, don't write anything
Write simple, 1-value arrays in [...] format

### CODE REVIEW

File: mono_repo/lib/src/yaml.dart [hunk 1]
```diff
@@ -178,6 +201,15 @@ void _writeYaml(
         buffer.write(' ');
       }
       buffer.write('[]');
+    } else if (source.length == 1 && _isSimple(source.single)) {
```
Comment (jakemac53) [inline mono_repo/lib/src/yaml.dart:204]: I am not sure this is really worth maintaining or even desirable? I personally wouldn't do this in hand written code in the first place (I think its better to have consistent style for yaml, and I much prefer not using any json inherited syntax).

**Comment by jakemac53** [comment]
Honestly I sort of have the same question regarding strings, why not just quote all of them?

File: mono_repo/lib/src/yaml.dart (hunk 1 above)
Comment (kevmoo) [reply mono_repo/lib/src/yaml.dart:204]: Understood. I think it's kinda nice to make more "conventional" output. But I can nix this one.

Comment (kevmoo) [reply mono_repo/lib/src/yaml.dart]: Reverted these bits. PTAL

**Review by jakemac53 — APPROVED** [review]

**Pull request merged**